### ISSUE TITLE
Incorrect code in recent change

### ISSUE THREAD

**Issue by original-birdman** [OpenPLi/enigma2#1126]
https://github.com/OpenPLi/enigma2/commit/44e54847ee6e37f03b9a5784bdeeda078523e7dd

This code is checking for whether a tuner/frontend is a USB one.
But it's looking at the wrong symlink.
It should be looking at the `..../dvb%d.frontend0/device/subsystem` to find out which system bus it is on (the last item in the link value)
This is a method which actually works for all tuners (it will return "platform" for builtin ones).
The correct logic, based on who `libsysfs` looks at things (and simpler code - self contained at the single point of usage) exists at:

https://github.com/OpenViX/enigma2/commit/4afdd4122cc90cb9308c386cc1f45f1c3bb39b95

**Comment by athoik** [comment]
As explained in my commit, we are resolving the `/sys/class/dvb/dvb1.frontend0` and if resolved path contains `usb` we assume that device is a usb one.

On the other hand, resolving the `/sys/class/dvb/dvb1.frontend0/device/subsystem` seems **more accurate** because it always contains `/usb` at the end.

I do not see something wrong, in my original commit, neither knew your "version" of `isusb`. Both are doing the same!

Most probably adding some wrappers for contains,endsWith,startsWith (in order to reuse then in many places where C code is now used) and check subsystem will be fine, what do you think?

```
diff --git a/lib/base/wrappers.cpp b/lib/base/wrappers.cpp
index 2073233..65a2cc3 100644
--- a/lib/base/wrappers.cpp
+++ b/lib/base/wrappers.cpp
@@ -310,3 +310,18 @@ std::string readLink(const std::string &link)
        ssize_t size = ::readlink(link.c_str(), buf, sizeof(buf));
        return std::string(buf, (size > 0) ? size : 0);
 }
+
+bool contains(const std::string &str, const std::string &substr)
+{
+        return substr.size() && str.size() >= substr.size() && str.find(substr) != std::string::npos;
+}
+
+bool endsWith(const std::string &str, const std::string &suffix)
+{
+        return suffix.size() && str.size() >= suffix.size() && str.find(suffix) + suffix.size() == str.size();
+}
+
+bool startsWith(const std::string& str, const std::string& prefix)
+{
+        return prefix.size() && str.size() >= prefix.size() && str.find(prefix) == 0;
+}
diff --git a/lib/base/wrappers.h b/lib/base/wrappers.h
index 74739f2..1e2ce77 100644
--- a/lib/base/wrappers.h
+++ b/lib/base/wrappers.h
@@ -10,5 +10,8 @@ int Connect(const char *hostname, int port, int timeoutsec);
 std::string base64encode(const std::string str);
 std::string base64decode(const std::string hash);
 std::string readLink(const std::string &link);
+bool contains(const std::string &str, const std::string &substr);
+bool endsWith(const std::string &str, const std::string &suffix);
+bool startsWith(const std::string &str, const std::string &prefix);

 #endif
diff --git a/lib/dvb/dvb.cpp b/lib/dvb/dvb.cpp
index 73bea6c..3d8824e 100644
--- a/lib/dvb/dvb.cpp
+++ b/lib/dvb/dvb.cpp
@@ -248,8 +248,8 @@ bool eDVBAdapterLinux::isusb(int nr)
        {
                return true;
        }
-       snprintf(devicename, sizeof(devicename), "/sys/class/dvb/dvb%d.frontend0", nr);
-       return readLink(devicename).find("usb") != std::string::npos;
+       snprintf(devicename, sizeof(devicename), "/sys/class/dvb/dvb%d.frontend0/device/subsystem", nr);
+       return endsWith(readLink(devicename), "/usb");
 }

 DEFINE_REF(eDVBUsbAdapter);
```

**Comment by original-birdman** [comment]
The _correct_ thing to do is to check for the bus that the device is on.  Which, as I've noted, is what `libsysfs` does for the equivalent functionality (it's what I looked at to ensure I got the correct test). And, as a general rule,  "correct" is much less likely to cause problems in the future.

And writing extra functions to create a short-lived C++-string object just to check the end of a C-string (which is all that the system call is ever going to return, and what you already have) seems to be overkill.
